src/js/chat/handlers.js

```diff
@@ -523,6 +523,9 @@ var privmsg = exports.privmsg = function(channel, data) {
         from = data.bttvDisplayName;
     } else {
         from = helpers.lookupDisplayName(data.from);
+        if (bttv.settings.get('disableLocalizedNames') === true && from.toLowerCase() !== data.from) {
```

Comment:
You are right that putting it inside there is cleaner, but I'm fairly sure my fix didn't break nicknames, as those happen in the other case of the if statement, I think?